tests/test_relativedeltafield.py

```diff
@@ -232,3 +234,16 @@ def test_value_usable_as_timedelta(self):
         obj1 = Interval(value='P1Y2M3W4DT5H6M7S')
         sum = datetime(2019, 1, 4, 10, 11, 12, 500) + obj1.value
         self.assertEqual(sum, datetime(2020, 3, 29, 15, 17, 19, 500))
+
+    # def test_simple_annotation(self):
+    #     one_month = Cast(
+    #         Value(relativedelta(months=1), output_field=RelativeDeltaField()),
+    #         DurationField()
+    #     )
+    #     q = Interval.objects.annotate(month_earlier=F('date') - one_month)
+    #     try:
+    #         str(q.query)
+    #     except AttributeError as e:
+    #         pytest.fail("Should not raise AttributeError")
+
```

Comment:
Doesn't make a lot of sense to add a test and then comment it out :)

Reply:
This is where you need to fill me in... I'm not clear how you'd like the code fragment you provided to work...

Reply:
Ah, I see.  It's always a bit messy (I really struggle with Django's ORM sometimes), but this piece of code seems to work:

```
        one_month = Value(relativedelta(months=1))
        q = Interval.objects.annotate(month_earlier=ExpressionWrapper(F('date') - one_month, output_field=RelativeDeltaField()))
```

I would suggest making it a more functional test that actually tests the filter is working:

```
    @pytest.mark.xfail('postgresql' not in settings.DATABASES['default']['ENGINE'], reason="Only works with PostgreSQL")
    def test_simple_annotation(self):
        interval = Interval(date='2020-10-06', value=None)
        interval.save()

		# This Value/Cast dance is required because psycopg2 doesn't
		# know how to adapt RelativeDelta literals.
        one_month = Cast(
            Value(relativedelta(months=1), output_field=RelativeDeltaField()),
            DurationField()
        )
        q = Interval.objects.annotate(month_earlier=ExpressionWrapper(F('date') - one_month, output_field=DateField()))
        self.assertEqual(0, q.filter(month_earlier__lt='2020-09-06').count())
        self.assertEqual(1, q.filter(month_earlier__lte='2020-09-06').count())
```

Bonus points if you can somehow make it work with MySQL! :)

Reply:
Hi... finally posted a working test version. Used pytest with fixtures to create mocked values.
No bonus I am afraid for making it work with that lesser db that is mysql ( :-) ) ... Maybe in a next release ...